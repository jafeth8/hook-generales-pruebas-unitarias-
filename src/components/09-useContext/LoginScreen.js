import React, { useContext } from 'react'
import { UserContext } from './UserContex'

export const LoginScreen = () => {
    const {setUser} = useContext(UserContext)
    const addUser=()=> {
        setUser({user:1234,name:'el prro'})
    }
    return (
        <div>
            <h1>Login Screen</h1>
            <hr/>
            
            <button className="btn btn-primary" onClick={addUser}>
                nuevo usuario
            </button>
        </div>
    )
}
