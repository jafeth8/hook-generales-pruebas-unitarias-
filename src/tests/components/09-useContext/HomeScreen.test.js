import {mount} from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContex';



describe('pruebas en el componente <HomeScreen/>', () => {

    const user={
        name:'Areli',
        email: 'areli@gmail.com'
    }

    const wrapper = mount(
        <UserContext.Provider value={{ user }}>
            <HomeScreen/>
        </UserContext.Provider>
        
    );

    test('debe mostrarse correctamente', () => {
       expect(wrapper).toMatchSnapshot();

       let pre =wrapper.find('pre')
       console.log(pre.html());
       expect(pre.text().trim()).toBe(JSON.stringify(user,null,3))
    })
    
})
