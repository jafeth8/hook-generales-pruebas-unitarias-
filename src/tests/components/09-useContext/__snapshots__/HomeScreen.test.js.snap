// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente <HomeScreen/> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Areli",
   "email": "areli@gmail.com"
}
  </pre>
</div>
`;
